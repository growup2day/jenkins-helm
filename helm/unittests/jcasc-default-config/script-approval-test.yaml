templates:
  - config-jcasc.yaml
tests:
  - it: should include script approval config in default config scripts when configured
    set:
      JCasC:
        defaultConfig: true
        scriptApproval:
          - "method groovy.json.JsonSlurperClassic parseText java.lang.String"
          - "new groovy.json.JsonSlurperClassic"
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-script-approval.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-script-approval.yaml]
  - it: should include blank script approval config in default config scripts when not configured
    set:
      JCasC:
        defaultConfig: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-script-approval.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-script-approval.yaml]
  - it:
      should not include script approval config in default config scripts when there is custom config script for the
      section
    set:
      JCasC:
        defaultConfig: true
        configScripts:
          custom-override: |
            security:
              scriptApproval:
                approvedSignatures: []
    asserts:
      - isNull:
          path: data.[jcasc-default-config-script-approval.yaml]
      - isNotNull:
          path: data.[custom-override.yaml]
