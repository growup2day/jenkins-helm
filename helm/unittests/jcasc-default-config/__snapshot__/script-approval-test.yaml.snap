should include blank script approval config in default config scripts when not configured:
  1: |
    |-
      security:
        # In-process Script Approval
        scriptApproval:
          approvedSignatures: []
should include script approval config in default config scripts when configured:
  1: |
    |-
      security:
        # In-process Script Approval
        scriptApproval:
          approvedSignatures:
              - "method groovy.json.JsonSlurperClassic parseText java.lang.String"
              - "new groovy.json.JsonSlurperClassic"
